ihontaryk/homework_10/iterators.py

```diff
@@ -0,0 +1,61 @@
+class Iterator:
+    def __init__(self, iterable):
+        self.iterable = iterable
+        self.index = 0
+
+    def __iter__(self):
+        return self
+
+
+class SimpleIterator(Iterator):
+    def __next__(self):
+        if self.index >= len(self.iterable):
+            raise StopIteration
+
+        element = self.iterable[self.index]
+        self.index += 1
+
+        return element
+
+
+class ReversedOperator(Iterator):
+    def __init__(self, iterable):
+        super().__init__(iterable)
```

Comment:
sometimes it is better to copy but not inherrit, since inherritance is not always obvious and easy to track

Reply:
![Screenshot 2024-11-09 173222](https://github.com/user-attachments/assets/0b915370-bfa8-40f0-ade8-21a766edf5b8)